crates/bevy_window/src/window.rs

```diff
@@ -383,6 +404,8 @@ pub struct WindowDescriptor {
     pub cursor_visible: bool,
     pub cursor_locked: bool,
     pub mode: WindowMode,
+    #[cfg(any(target_os = "windows", target_os = "linux"))]
```

Comment:
Personally, I'd be happier to have the icon be silently ignored on platforms where it isn't supported.
In its current form this is a feature gate footgun which makes it less likely that applications would support e.g. darwin or android out of the gate.

Additionally, it seems reasonable that we might in future support making this change the website favicon, perhaps behind a feature gate

```suggestion
    /// The icon for the window. Only active on platforms which support it, which is currently Windows and X11 Linux.
```